packages/core/src/Synchronizer.ts

```diff
@@ -10,57 +10,157 @@ import { constructSchema } from 'anondb/types'
 import { MemoryConnector } from 'anondb/web'
 import AsyncLock from 'async-lock'
 
+/**
+ * The type of data to be processed in an event handler.
+ */
 type EventHandlerArgs = {
     event: ethers.Event
     decodedData: { [key: string]: any }
     db: TransactionDB
 }
 
+/**
+ * The settings of an attester.
+ */
 type AttesterSetting = {
     startTimestamp: number
     epochLength: number
 }
 
+/**
+ * Turn a either decimal or hex string to decimal string.
+ * @param content A `bigint`, `string` or `number` type data.
+ * @returns A decimal string.
+ */
 export function toDecString(content: bigint | string | number) {
     return BigInt(content).toString()
 }
 
 /**
  * The synchronizer is used to construct the Unirep state. After events are emitted from the Unirep contract,
  * the synchronizer will verify the events and then save the states.
+ * @example
+ * ```ts
+ * import { Synchronizer } from '@unirep/core'
+ *
+ * const state = new Synchronizer({
+ *   unirepAddress: '0xaabbccaabbccaabbccaabbccaabbccaabbccaaaa',
+ *   provider, // an ethers.js provider
+ * })
+ *
+ * // start the synchronizer deamon
+ * await state.start()
+ * await state.waitForSync()
+ *
+ * // stop the synchronizer deamon
+ * state.stop()
+ * ```
  */
 export class Synchronizer extends EventEmitter {
+    /**
+     * @private
+     * The database object.
+     */
     private _db: DB
+    /**
+     * @private
+     * The provider which is connected in the synchronizer.
+     */
     private _provider: ethers.providers.Provider
+    /**
+     * @private
+     * The UniRep smart contract object which is connected in the synchronizer.
+     */
     private _unirepContract: ethers.Contract
+    /**
+     * @private
+     * The array of attester IDs which are synchronized in the synchronizer.
+     */
     private _attesterId: bigint[] = []
+    /**
+     * @private
+     * The circuits settings.
+     */
     private _settings: any
+    /**
+     * @private
+     * The settings of attesters. There are `startTimestamp` and `epochLength` of each attester.
+     */
     private _attesterSettings: { [key: string]: AttesterSetting } = {}
+    /**
+     * @protected
+     * The default zero [state tree](https://developer.unirep.io/docs/protocol/trees#state-tree) leaf.
+     */
     protected defaultStateTreeLeaf: bigint = BigInt(0)
+    /**
+     * @protected
+     * The default zero [epoch tree](https://developer.unirep.io/docs/protocol/trees#epoch-tree) leaf.
+     */
     protected defaultEpochTreeLeaf: bigint = BigInt(0)
+    /**
+     * @private
+     * Indicate the synchronizer if to sync all UniRep attesters.
```

Comment:
'Indicate if the synchronizer is to sync with all Unirep attesters.'  ??

Reply:
yes!